torch/utils/cpp_extension.py

```diff
@@ -488,19 +507,34 @@ def _check_cuda_version(compiler_name: str, compiler_version: TorchVersion) -> N
             )
 
 
+def _append_sycl_std_if_no_std_present(cflags):
+    if not any(flag.startswith('-sycl-std=') for flag in cflags):
+        cflags.append('-sycl-std=2020')
+
+
+def _wrap_sycl_host_flags(cflags):
+    host_cxx = get_cxx_compiler()
+    host_cflags = [
+        f'-fsycl-host-compiler={host_cxx}',
+        shlex.quote(f'-fsycl-host-compiler-options={cflags}'),
```

Comment:
Refer to https://github.com/intel/llvm/blob/1e3940b05297795d2584fccc7f0de2a3152dec09/sycl/doc/UsersManual.md?plain=1#L385, `opts` should be quoted as a string instead of the whole statement.
```suggestion
        f'-fsycl-host-compiler-options="{cflags}",
```

Reply:
@guangyey, unfortunately this won't work. I tried your change - it causes compilation failure. Actually I have spent significant time figuring out how to make this place working. What I currently have in the PR is the working variant. Other variants did not work for me. The reason why this is complicated is that we need to pass quoted arguments inside a quoted string. Something like:
```
-fsycl-host-compiler-options="-Wno-unused-result -DPYBIND11_COMPILER_TYPE=\"_gcc\" "
```
And the way dpc++ compiler is implemented complicates this out (due to having a shell wrapper). By fact we have few layers which attempt to drop quotes. As a result we need to heavily escape them.

Note that ultimate goal is to pass `-fsycl-host-compiler-options=str` argument as a single argument to dpc++. Quotes in the documentation is just one way to achieve that. Another way is to quote entire argument and yet another way would be to escape spaces. So, PR achieves the goal by quoting the whole argument, i.e. we pass:
```
<...> -fsycl-host-compiler=c++  '-fsycl-host-compiler-options=-Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/home/dvrogozh/git/pytorch/pytorch/torch/include -I/home/dvrogozh/git/pytorch/pytorch/torch/include/torch/csrc/api/include -I/home/dvrogozh/git/pytorch/pytorch/test/cpp_extensions/self_compiler_include_dirs_test -I/home/dvrogozh/pytorch.xpu/include -I/usr/include/python3.10 -c -g -DTORCH_API_INCLUDE_EXTENSION_H -DPYBIND11_COMPILER_TYPE=\\"_gcc\\" -DPYBIND11_STDLIB=\\"_libstdcpp\\" -DPYBIND11_BUILD_ABI=\\"_cxxabi1016\\" -DTORCH_EXTENSION_NAME=sycl -D_GLIBCXX_USE_CXX11_ABI=1 -std=c++17' <...> 
```

See also https://github.com/intel/llvm/issues/15016 on that.

Reply:
I see. Thanks for your elaboration.